etrago/tools/calc_results.py

```diff
@@ -196,121 +209,565 @@ def calc_marginal_cost(self):
+                marginal_cost += (
+                    c.pnl.p.mul(c.df.marginal_cost)
+                    .mul(network_de.snapshot_weightings.generators, axis=0)
+                    .sum()
+                    .sum()
+                )
+
+        else:
+            if "marginal_cost" in c.df.columns:
+                marginal_cost += (
+                    c.pnl.p0.mul(c.df.marginal_cost)
+                    .mul(network_de.snapshot_weightings.generators, axis=0)
+                    .sum()
+                    .sum()
+                )
+        if c.name not in [
+            "Bus",
+            "Load",
+            "LineType",
+            "TransformerType",
+            "Carrier",
+        ]:
+            invest_cost += (
+                (
+                    c.df[c.df[f"{value}_nom_extendable"]][f"{value}_nom_opt"]
+                    - c.df[c.df[f"{value}_nom_extendable"]][f"{value}_nom_min"]
+                )
+                * c.df[c.df[f"{value}_nom_extendable"]]["capital_cost"]
+            ).sum()
+
+    # import and its costs
+    links_export = self.network.links[
+        (
+            self.network.links.bus0.isin(network_de.buses.index.values)
+            & ~(self.network.links.bus1.isin(network_de.buses.index.values))
+        )
+    ]
+
+    export_positive = (
+        self.network.links_t.p0[links_export.index]
+        .clip(lower=0)
+        .mul(self.network.snapshot_weightings.generators, axis=0)
+        .mul(
+            self.network.buses_t.marginal_price[links_export.bus1].values,
+        )
+        .sum()
+        .sum()
+    )
+
+    export_negative = (
+        self.network.links_t.p0[links_export.index]
+        .clip(upper=0)
+        .mul(self.network.snapshot_weightings.generators, axis=0)
+        .mul(
+            self.network.buses_t.marginal_price[links_export.bus1].values,
+        )
+        .mul(-1)
+        .sum()
+        .sum()
+    )
+
+    links_import = self.network.links[
+        (
+            self.network.links.bus1.isin(network_de.buses.index.values)
+            & ~(self.network.links.bus0.isin(network_de.buses.index.values))
+        )
+    ]
+
+    import_positive = (
+        self.network.links_t.p0[links_import.index]
+        .clip(lower=0)
+        .mul(self.network.snapshot_weightings.generators, axis=0)
+        .mul(
+            self.network.buses_t.marginal_price[links_import.bus1].values,
+        )
+        .sum()
+        .sum()
+    )
+
+    import_negative = (
+        self.network.links_t.p0[links_import.index]
+        .clip(upper=0)
+        .mul(self.network.snapshot_weightings.generators, axis=0)
+        .mul(
+            self.network.buses_t.marginal_price[links_import.bus1].values,
+        )
+        .mul(-1)
+        .sum()
+        .sum()
+    )
+
+    import_costs = (
+        export_negative + import_positive - export_positive - import_negative
+    )
+
+    return marginal_cost, invest_cost, import_costs
```

Comment:
It looks like the function works well. Once system_costs_germany() is called, a tuple of 3 values is shown. Could you include in the output of this function what each of them represents?

Reply:
The returned values are already described within the docstring. Do you mean that they should be described somewhere else?